common-authentication/src/main/java/org/alfresco/hxi_connector/common/adapters/messaging/repository/ApplicationInfoProvider.java

```diff
@@ -52,8 +52,10 @@ public String getUserAgentData()
     {
         if (StringUtils.isBlank(applicationInfo))
         {
+            log.debug("Calculating user agent data.");
             applicationInfo = calculateUserAgentData();
         }
+        log.atDebug().log("User agent data: {}", applicationInfo);
```

Comment:
To be precise both usages of `log.debug` / `log.atDebug` in this PR could have been `log.debug` instead because there is no additional processing done in these cases.

`log.debug("Calculating user agent data.");` is just logging a static string
`log.atDebug().log("User agent data: {}", applicationInfo);` is just logging a string where `applicationInfo` has already been fully resolved via `calculateUserAgentData` and no further method invocations are being done on it _(e.g.: `applicationInfo.getSomething()`)_

Even then, I very much doubt all these concerns about performance are of any real significance, so we can choose to proceed as we see fit. If we think it's easier to just spam `log.atDebug().log(...)` even though it's unnecessary, then we can do it.

Reply:
Domenico is right about keeping `log.debug` as sufficient enough here - I can change that.
And I also doubt there is any significant performance implication.
BTW, I think PMD complained about not using `log.atDebug()` in cases where there actually is some additional processing - we should probably keep that usage to minimal, WDYT?

Reply:
I'd be ok with only using the more verbose `log.atDebug().log(...)` syntax only when actually useful.

Reply:
_(PMD should spot that)_

Reply:
Fixed  - only `log.debug` is used now.